### PR TITLE
Added Named SoundEvents to Sound Effect Packets

### PR DESCRIPTION
**Pull request by OfficialKris**
## Description
Removed comments in favor of supporting sound events by name. Renamed sound_id to sound_event as it is in MC (can be either ID or SoundEvent). Allows for sounds with custom fixed distances.

## Testing
Ran the following code:
```rust
world.broadcast_packet_all(&CSoundEffect::new(
        VarInt(-1), // needed to send 0x00 to client to allow sound_name
        Some(SoundEvent {
            sound_name: String::from("ambient.cave"),
            range: Some(30.0),
        }),
        SoundCategory::Records,
        emit_sound_loc.x,
        emit_sound_loc.y,
        emit_sound_loc.z,
        1.0,
        1.0,
        seed,
    ))
    .await;
```

## Checklist
Things need to be done before this Pull Request can be merged.

- [x] Code is well-formatted and adheres to project style guidelines: `cargo fmt`
- [x] Code does not produce any clippy warnings `cargo clippy`

### CODE REVIEW

File: pumpkin-protocol/src/client/play/c_entity_sound_effect.rs [hunk 1]
```diff
@@ -1,16 +1,30 @@
 use pumpkin_macros::client_packet;
-use serde::Serialize;
 
-use crate::{SoundCategory, VarInt};
+use crate::{ClientPacket, IDOrSoundEvent, SoundCategory, SoundEvent, VarInt};
+
+impl ClientPacket for CEntitySoundEffect {
```
Comment (Snowiiii) [inline pumpkin-protocol/src/client/play/c_entity_sound_effect.rs:5]: We usually always implement ClientPacket or ServerPacket at the under the new impl. You should do that as well so the Code follows the existing code style

File: pumpkin-protocol/src/client/play/c_sound_effect.rs [hunk 2]
```diff
@@ -1,16 +1,32 @@
 use pumpkin_macros::client_packet;
-use serde::Serialize;
 
-use crate::{SoundCategory, VarInt};
+use crate::{ClientPacket, IDOrSoundEvent, SoundCategory, SoundEvent, VarInt};
+
+impl ClientPacket for CSoundEffect {
```
Comment (Snowiiii) [inline pumpkin-protocol/src/client/play/c_sound_effect.rs:5]: Here as well

File: pumpkin-protocol/src/lib.rs [hunk 3]
```diff
@@ -68,6 +68,18 @@ pub enum SoundCategory {
     Voice,
 }
 
+#[derive(Serialize)]
+pub struct IDOrSoundEvent {
+    pub id: VarInt,
+    pub sound_event: Option<SoundEvent>,
+}
```
Comment (Snowiiii) [inline pumpkin-protocol/src/lib.rs:75]: Im not sure what this is Struct is. Looking at Vanilla code the Packets do just need the SoundEvent

File: pumpkin-protocol/src/client/play/c_entity_sound_effect.rs (hunk 1 above)
Comment (OfficialKris) [reply pumpkin-protocol/src/client/play/c_entity_sound_effect.rs:5]: I was following the docs, it makes it seem to replace #[derive(serialize)] with the impl ClientPacket. Maybe a note in the documentation could be added?

File: pumpkin-protocol/src/lib.rs (hunk 3 above)
Comment (OfficialKris) [reply pumpkin-protocol/src/lib.rs:75]: This Struct is an implementation from the wiki.vg page. However, the wiki.vg page seems to be wrong in multiple locations for these packets, they did update it recently but it still wrong. Maybe this part is wrong as well. I will give it a look.

Comment (OfficialKris) [reply pumpkin-protocol/src/lib.rs:75]: Just looked at the vanilla code and it seems like it only sends SoundEvent but it must be adding an additional zero hex while encoding the object. Without the extra 0x00 sent, the client does not accept the additional string bytes. The vanilla codebase must only use the strings and thus the additional id parameter is not necessary for them to implement. We can do the same and just remove this struct in favor of a encoded 0x00.

**Comment by Snowiiii** [comment]
Overall looks good to me.
Thanks @OfficialKris

**Pull request merged**